File: starters/java/src/main/java/io/fproto/core/SessionConfig.java
package io.fproto.core;

public record SessionConfig(
    int idleTimeoutSecs,
    int expiryTimeoutSecs,
    int handshakeTimeoutSecs,
    int maxConnections
) {
    public static SessionConfig defaults() {
        return new SessionConfig(300, 86400, 30, 10000);
    }
}

File: starters/php/src/Session/SessionConfig.php
<?php

declare(strict_types=1);

namespace FProto\Session;

final readonly class SessionConfig
{
    public function __construct(
        public int $idleTimeoutSecs = 300,
        public int $expiryTimeoutSecs = 86400,
        public int $handshakeTimeoutSecs = 30,
        public int $maxConnections = 10000,
    ) {}
}
